target/avr/helper.c

```diff
@@ -24,17 +24,18 @@
 #include "exec/address-spaces.h"
 #include "exec/helper-proto.h"
 
+static void avr_cpu_do_interrupt_locked(CPUState *cs);
+
 bool avr_cpu_exec_interrupt(CPUState *cs, int interrupt_request)
 {
     bool ret = false;
-    CPUClass *cc = CPU_GET_CLASS(cs);
     AVRCPU *cpu = AVR_CPU(cs);
     CPUAVRState *env = &cpu->env;
 
     if (interrupt_request & CPU_INTERRUPT_RESET) {
         if (cpu_interrupts_enabled(env)) {
             cs->exception_index = EXCP_RESET;
-            cc->do_interrupt(cs);
+            avr_cpu_do_interrupt_locked(cs);
```

Comment:
This is different from previous patterns.
I would expect:
AVRCPU *avr_cpu = AVR_CPU(cs);
avr_cpu->do_interrupt_locked(cs);